### PR TITLE
Make deployStaging a separate job in preDeploy workflow

### PR DESCRIPTION
**Pull request by roryabraham**
The purpose of this PR is to make sure that the `staging` (and soon `production`) branches are always kept up-to-date, and that we only ever deploy to staging and/or production when the corresponding branch has been successfully updated. Before, we were deploying, then creating an automerge PR to update the branch. Now, we'll create an automerge PR to update the branch, then deploy, which is safer & makes more sense.

If also allows us to "escalate" changes so that we can effectively propagate changes like:

1) PR is merged to master
2) Update the version on master
3) Update the version on staging by creating a PR from master -> staging
4) Staging is updated, so deploy staging
5) StagingDeployCash is closed, update production by creating a PR from staging -> production
6) Production is updated, so deploy production

### Fixed Issues
Fixes https://github.com/Expensify/Expensify/issues/155208

### Tests
This might be a merge-to-test, but we may be able to do some testing in Public-Test-Repo first

### CODE REVIEW

File: .github/workflows/deploy.yml [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    steps:
+      - name: Checkout staging
+        uses: actions/checkout@5a4ac9002d0be2fb38bd78e4b4dbde5606d7042f
+        with:
+          ref: staging
+          token: ${{ secrets.OS_BOTIFY_TOKEN }}
+
+      - name: Get current version
+        id: getVersion
+        uses: Expensify/Expensify.cash/.github/actions/getCurrentAppVersion@master
```
Comment (AndrewGable) [inline .github/workflows/deploy.yml:38]: Why not `npm run print-version --silent`?

File: .github/workflows/preDeploy.yml [hunk 2]
```diff
@@ -131,3 +126,30 @@ jobs:
...
+      && needs.chooseDeployActions.outputs.isStagingDeployLocked == 'false'
+    }}
+
+    steps:
+      - name: Create Pull Request
+        # Version: 2.4.3
+        uses: repo-sync/pull-request@33777245b1aace1a58c87a29c90321aa7a74bd7d
+        with:
+          source_branch: master
+          destination_branch: "staging"
```
Comment (AndrewGable) [inline .github/workflows/preDeploy.yml:144]: Can we just not use the " "?

File: .github/workflows/preDeploy.yml [hunk 3]
```diff
@@ -30,6 +31,9 @@ jobs:
         with:
           GITHUB_TOKEN: ${{ secrets.OS_BOTIFY_TOKEN }}
 
+      - name: Check if merged pull request was an automatic version bump PR
+        run: echo "::set-output name=IS_VERSION_BUMP_PR::${{ constains(steps.getMergedPullRequest.outputs.labels, automerge) && github.actor == 'OSBotify' }}"
```
Comment (AndrewGable) [inline .github/workflows/preDeploy.yml:35]: Are we expecting this to return `true`/`false`? Can we verify it works as intended via public repo? I have had issues with booleans

File: .github/actions/getCurrentAppVersion/action.yml [hunk 4]
```diff
@@ -0,0 +1,12 @@
+name: 'Get the current app version'
+description: 'Get the current app version from package.json'
+inputs:
+    GITHUB_TOKEN:
+        description: Auth token for Expensify.cash Github
+        required: true
```
[the remark this answers is not in the document]
Comment (roryabraham) [reply .github/actions/getCurrentAppVersion/action.yml:6]: Actually, it may not be needed.

File: .github/workflows/preDeploy.yml (hunk 2 above)
Comment (roryabraham) [reply .github/workflows/preDeploy.yml]: I think so. Removed!

File: .github/workflows/preDeploy.yml (hunk 3 above)
Comment (roryabraham) [reply .github/workflows/preDeploy.yml:35]: Tested in public repo w/ code:

```yaml
jobs:
  echoActor:
    runs-on: macos-latest
    outputs:
      isActorRory: ${{ steps.isActorRory.outputs.IS_ACTOR_RORY }}
    steps:
      - name: Echo actor
        run: echo ${{ github.actor }}

      - name: Is Actor Rory?
        id: isActorRory
        run: echo "::set-output name=IS_ACTOR_RORY::${{ github.actor == 'roryabraham' && 'blah' == 'blah' }}"

  # This workflow contains a single job called "build"
  build:
    # The type of runner that the job will run on
    runs-on: macos-latest
    needs: echoActor
    if: ${{ needs.echoActor.outputs.isActorRory == 'true' }}

    # Steps represent a sequence of tasks that will be executed as part of the job
    steps:
      - name: Run a nested script
        run: echo "Hello $(echo 'World')!"

      # Runs a single command using the runners shell
      - name: Run a one-line script
        run: echo ${{ toJSON(github.event) }} && echo "I am Andrew" && exit 0
```

And as you can see in the [workflow run](https://github.com/Andrew-Test-Org/Public-Test-Repo/actions/runs/658456018/workflow), the `build` job is not skipped.

Comment (roryabraham) [reply .github/workflows/preDeploy.yml:35]: We are comparing the result to string `'true'` and/or `'false'` throughout the rest of the workflow, as we've discovered is usually necessary

File: .github/workflows/preDeploy.yml [hunk 5]
```diff
@@ -30,10 +31,15 @@ jobs:
...
+
   skipDeploy:
     runs-on: ubuntu-latest
     needs: chooseDeployActions
-    if: ${{ needs.chooseDeployActions.outputs.isStagingDeployLocked == 'true' }}
+    if: |
```
Comment (roryabraham) [inline .github/workflows/preDeploy.yml:40]: Tested this syntax for multiline if statements [here](https://github.com/Andrew-Test-Org/Public-Test-Repo/runs/2124135546?check_suite_focus=true)

**Review by AndrewGable — APPROVED** [review]

**Comment by AndrewGable** [comment]
@deetergp - Feel free to review but in order to test I am going to merge this! 🚀

**Pull request merged**